src/metrics/fn_args.rs

```diff
@@ -100,3 +124,73 @@ impl NArgs for JavaCode {}
 impl NArgs for GoCode {}
 impl NArgs for CssCode {}
 impl NArgs for HtmlCode {}
+
+#[cfg(test)]
+mod tests {
+    use std::path::PathBuf;
+
+    use super::*;
+
+    #[test]
+    fn test_function_nargs() {
+        check_metrics!(
+            "def f(a, b):
+                 if a:
+                     return a",
+            "foo.py",
+            PythonParser,
+            nargs,
+            [(nargs, 2, usize)],
+            [(nargs_average, 2.0)] // 1 function
+        );
+    }
+
+    #[test]
+    fn test_functions_nargs() {
+        check_metrics!(
+            "def f(a, b):
+                 if a:
+                     return a
+            def f(a, b):
+                 if b:
+                     return b",
+            "foo.py",
+            PythonParser,
+            nargs,
+            [(nargs, 4, usize)],
+            [(nargs_average, 2.0)] // 2 functions
+        );
+
+        check_metrics!(
+            "def f(a, b):
+                 if a:
+                     return a
+            def f(a, b, c):
+                 if b:
+                     return b",
+            "foo.py",
+            PythonParser,
+            nargs,
+            [(nargs, 5, usize)],
+            [(nargs_average, 2.5)] // 2 functions
+        );
+    }
+
+    #[test]
+    fn test_nested_functions_nargs() {
+        check_metrics!(
+            "def f(a, b):
+                 def foo(a):
+                     if a:
+                         return 1
+                 bar = lambda a: lambda b: b or True or True
+                 return bar(foo(a))(a)",
+            "foo.py",
+            PythonParser,
+            nargs,
+            // FIXME: Consider lambda arguments also
```

Comment:
Please file an issue about this, so we don't forget.

Reply:
#342